### ISSUE TITLE
Atom Render viewport for animation editor

### ISSUE THREAD

**Issue by amzn-rhhong** [o3de/rfcs#45]
### Summary:
Replace OpenGLRenderViewport in emotionfx with Atom Render Viewport.

### What is the relevance of this feature?

Since EMotionFX released beta in LY 1.10, it has been using OpenGL to render actor in the animation editor. It has many disadvantages including:

1）Performance. We are still using outdated OpenGL rendering technology and creating actor format for OpenGL needs.

2)  Different rendering results with animation editor vs in game.

3)  An inconsistent workflow (compared to other o3de viewport) when navigating the OpenGL render viewport.

4)  OpenGL is no long supported on Mac. Currently EMFX's opengl renderviewport is the only renderer that's not integrated with Atom.

Integration to atom viewport solves all problems above and gives customer a much better experience in the animation editor. We plan to leverage some of the technology we have built for the UI editor and material editor (i.e Atom Viewport Context, Render Viewport Widget and Camera? work) to speed up development.

### Technical design description:

Here is a simplified version of the class diagram to help understand the current design for opengl render viewport and render plugin in emotionfx.

![currentDesign](https://user-images.githubusercontent.com/69218254/130696355-82fb1ee8-cb8a-4b3a-89cd-9bcc5192707c.png)

As you can see, it is going to be significant amount of job to replace MCore::Camera to AZ::Camera, MCore::Manipulator to AZ::Manipulator and keep supporting all the OpenGL rendering and functions on parallel to atom renderer. The good thing is we don't have to be, since the goal is to completely remove openGL dependency and replace it with atom. Our plan is to use the existing RenderViewportWidget (which is already integrated with AZ::Camera, controller, TickBus, and currently already working for MaterialEditor), and built it from the ground up by adding the functionality that needed by emotionfx. It should be kept entirely separate from the existing renderPlugin.

New design for AtomRenderPlugin
Here is a simplified version of the class diagram for the AtomRenderViewportWidget

![atomRenderView](https://user-images.githubusercontent.com/69218254/130696389-3eae7c42-2f73-495e-8e98-ad55f9eca984.png)

In this new diagram, you will noticed a few thing got changed. First, compare to the first graph, we got rid of a lot of rendering specific class and functions. This is because those function were deferred to being handled by RenderViewportWidget. Second, usage of MCore::Camera and MCore::Manipulator are completely gone and replaced by AZ::Camera and AzToolsFramework::Manipulator. In addition, we also get rid of the layout setting and multiple viewWidget for the new renderPlugin, because it is not a core feature and not having a lot of usage. For now, the atom render plugin will only have one active viewport.

### Update the ui menu options
<html>
<body>

UI option name |  
-- | --
Select | Keep.
Manipulator | Keep. This option toggles between translation, rotation and scale manipulators.
Layout | Remove. Layout option offers split layout and there isn't much of a usage of that anymore.
View Option: |  
Solid, Lighting, Backface Culling, Grid | Keep. Atom should already have option to toggle them on and off.
Bounding Box, skeleton, joint names, joint orientaion, bind Pose, Colliders | Keep. All of these option should be able to be replaced by auxGeom rendering from atom.
Wireframe, vertex normals, face normals and tangents (include bitangents). | Keep. Those option will require software skinning of the actor. We would like to get rid of CPU skinning as it is terribly slow and not efficient. It would be great if atom can provide support to render those debug option on GPU later in the pipeline, then in here we could just enable the rendering options.The fallback plan is to still use software skinning and use auxGeom to render those options.
Gradient background | Remove. Not essential. In Material editor it support different skybox, if needed we could introduce that option instead.
Motion extraction | Keep. MCore render util uses a simple trajectory path class to render those and should be easily port over. The rendering function just combines some triangle and line rendering, should be easily replaced with auxGeom under the hood.
Camera Options: |  
Perspective, front, back, top, bottom, left, right | Keep. Checked with Atom team and editor team that orthographic camera is supported.
Reset Camera | Keep. Also there seems to be a bug that reset camera will reset the camera at a different distance vs clicking the camera position option (left, right, top..). Make sure that problem is addressed as well.
Show selected, show entire scene | Remove. Not essential. Almost never you would need multiple character on the animation editor, so that makes the option redundant.
Follow character | Keep.
Right Click: |  
Reset Transform, Open Actor, Recent Actor, Open workspace, Recent workspace, Transform, Camera options | Leaning Keep. Those are some of the essential UI options. Some might argue those are the same option can be found in file menus and it could be redundant. Could be an optional task and less prioritized.
Unselect/Hide/Unhide actor instances, Clone / Remove select actor instances, Merge Actor | Leaning Remove. Those option are not very often used as we don't recommend having multiple actor open in the animation editor. To me it's just add a bit confusion.

</body>
</html>

### Execution plan:

Here is a brief execution plan to show the order of the task should be work on and which of the task could be parallel between multiple SDEs. The openGLRenderPlugin redcode and emotionfx::Mesh redcode will be discussed after 1st release and could be done in parallel as awell. 

![executionplan](https://user-images.githubusercontent.com/69218254/130696416-437beaf8-647a-4fce-9e7a-2c3fe361f7ba.png)

### What are the advantages of the feature?
It will allow us to completely get rid of the OpenGL dependency. It will give us performance advantages using atom renderer and more align to other editor tools. The EMotionFX::Mesh format can also be deleted which would result in better performance in tool time.

### What are the disadvantages of the feature?
EMotionFX is further coupled with atom renderer. 

### How will this be implemented or integrated into the O3DE environment?
EMotionFX is already depended on Atom and its mesh format. User can just turn on the atom render viewport window instead of the openGL window in emotionfx editor.

### Are there any alternatives to this feature?
1) Keep the OpenGL render viewport.

2) If there is an alternative to atom renderer that is also a possibility by nature. But I haven't heard of any alternative and I'm sure no one want to go back to use Cry Renderer.

### How will users learn this feature?
The atom render viewport is designed to be a less complicated version of the openGL render viewport - and the interface will largely remain the same.

### Are there any open questions?
- Material still won't be rendered in the emotionfx editor. This is because currently emotionfx doesn't support the entity / component system yet and there isn't any place to do the material authoring. But if we want to support render material in the future, this would be a necessary step.
